drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter.java

```diff
@@ -0,0 +1,1540 @@
+            }
+            final DTCellValue52 modelCell = cd.getValue();
+            final BaseColumn modelColumn = model.getExpandedColumns().get( targetColumnIndex );
+            final DataType.DataTypes modelColumnDataType = columnUtilities.getDataType( modelColumn );
+            if ( cd.getValue().getDataType().equals( modelColumnDataType ) ) {
+                if ( modelCell.hasValue() ) {
+                    uiModel.setCell( targetRowIndex,
+                                     targetColumnIndex,
+                                     gridWidgetCellFactory.convertCell( modelCell,
+                                                                        modelColumn,
+                                                                        cellUtilities,
+                                                                        columnUtilities ) );
+                } else {
+                    uiModel.deleteCell( targetRowIndex,
+                                        targetColumnIndex );
+                }
+            }
+            if ( modelColumn instanceof RowNumberCol52 ) {
+                updateSystemControlledValues = true;
+            }
+        }
+        if ( updateSystemControlledValues ) {
+            synchronizer.updateSystemControlledColumnValues();
+        }
+        view.batch();
+    }
+
+    @Override
+    public void onDeleteSelectedCells() {
+        assertNotReadOnly();
+        final List<IGridData.SelectedCell> selections = uiModel.getSelectedCells();
+        if ( selections.size() == 0 ) {
+            return;
+        }
+        for ( IGridData.SelectedCell sc : selections ) {
+            final int rowIndex = sc.getRowIndex();
+            final int columnIndex = findUiColumnIndex( sc.getColumnIndex() );
+            final BaseColumn column = model.getExpandedColumns().get( columnIndex );
+            if ( !( column instanceof RowNumberCol52 ) ) {
+                uiModel.deleteCell( rowIndex,
+                                    columnIndex );
+            }
+        }
+        view.getLayer().draw();
+    }
+
+    @Override
+    public void onDeleteSelectedColumns() {
+        assertNotReadOnly();
+        final Set<Integer> selectedColumnIndexes = getSelectedColumnIndexes();
+        final Set<BaseColumn> columnsToDelete = new HashSet<>();
+        for ( int selectedColumnIndex : selectedColumnIndexes ) {
+            final int columnIndex = findUiColumnIndex( selectedColumnIndex );
+            final BaseColumn column = model.getExpandedColumns().get( columnIndex );
+            if ( column instanceof AttributeCol52 ) {
+                columnsToDelete.add( column );
+            } else if ( column instanceof MetadataCol52 ) {
+                columnsToDelete.add( column );
+            } else if ( column instanceof ConditionCol52 ) {
+                columnsToDelete.add( column );
+            } else if ( column instanceof ActionCol52 ) {
+                columnsToDelete.add( column );
+            }
```

Comment:
This block of `else if`s might be nicer with just one if (with several `OR` conditions), since the code for each branch is exactly the same.